WasatchNET/BoulderSpectrometer.cs

```diff
@@ -542,7 +593,60 @@ public override double[] getSpectrum(bool forceNew = false)
 
         }
 
-        protected override double[] getSpectrumRaw(bool skipTrigger=false)
+        public override async Task<double[]> getSpectrumAsync(bool forceNew = false)
+        {
+            if (!commError)
+            {
+                double[] sum = await getSpectrumRawAsync();
+                if (sum == null)
+                {
+                    logger.error("getSpectrum: getSpectrumRaw returned null");
+                    return null;
+                }
+                logger.debug("getSpectrum: received {0} pixels", sum.Length);
+
+                if (scanAveraging_ > 1)
+                {
+                    // logger.debug("getSpectrum: getting additional spectra for averaging");
+                    for (uint i = 1; i < scanAveraging_; i++)
+                    {
+                        double[] tmp = await getSpectrumRawAsync();
+                        if (tmp == null)
+                            return null;
+
+                        for (int px = 0; px < pixels; px++)
+                            sum[px] += tmp[px];
+                    }
+
+                    for (int px = 0; px < pixels; px++)
+                        sum[px] /= scanAveraging_;
+                }
+
+                if (dark != null && dark.Length == sum.Length)
+                    for (int px = 0; px < pixels; px++)
+                        sum[px] -= dark_[px];
+
+                if (correctPixels)

```

Comment:
I'd like to rename this to "correctBadPixels".  "correctPixels" feels like it could refer to a great many things in our post-processing.

Reply:
VS does not like the aliasing of having the same name refer to two things, I will change it though